airflow/models/dag.py

```diff
@@ -312,6 +316,7 @@ class DAG(LoggingMixin):
         ``Environment`` is used to render templates as string values.
     :param tags: List of tags to help filtering DAGs in the UI.
     :param schedule_on: List of upstream datasets if for use in triggering DAG runs.
+    :param auto_register: Automatically register this DAG when it is used in a ``with`` block
```

Comment:
If this is only applicable to context manager calls I think this is not the correct place to pass it or? Wouldn't it make more sense to define this on config level for all DAGs using context manager i.e. in the config?

Reply:
It's only relevant to context manager yes, but this is the only place you can add a flag to change the behavoiur of _one_ dag.

I think we could add a new config flag (default to True) that globally sets the auto_register behaviour, but I think we still need the way to change the behaviour for a single dag.

The other option I can think of is this:

```python
with DAG(dag_id=...).with_autoregister(False):
    ...
```

I don't have a strong preference either way

Reply:
Yeah. I like it the way it is now. with X.with_something() does not look good, and the current way is more explicit.